Added MailPoet upsell to email management page

fredrikekelund opened the pull request:
#### Proposed Changes

This PR adds an upsell link for the MailPoet plugin to the email management home screen (`/email/:site_slug`). The link leads to the marketplace page for the MailPoet plugin (`/plugins/mailpoet/:site_slug`) and records a Tracks event when clicked.

It's worth noting that because this is shown on the `/email/:site` page, it will only be visible to users who don't already have an email subscription or to users with multiple domains (because users with a single domain that has email will be automatically redirected from `/email/:site` to `/email/:domain/manage/:site`).

**EDIT:** I've replaced the original screenshot with the latest version.

![mailpoet-upsell](https://user-images.githubusercontent.com/1101677/183841290-256ca15b-5c60-43da-87b1-6e2b00d558c9.png)

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Have a site with one custom domain without any existing email subscription, or have a site with multiple custom domains (with or without email subscriptions)
2. Go to Upgrades > Emails
3. Ensure that the MailPoet upsell is displayed as per the screenshot above
4. Ensure that the link leads to the correct destination
5. Ensure that a Tracks event is recorded when clicking the link

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

fredrikekelund committed:
Added MailPoet upsell to EmailHome

fredrikekelund committed:
Inline MailPoet plugin URL

fredrikekelund committed:
Translatable section heading

stephanethomas commented on client/my-sites/email/email-management/home/mailpoet-upsell.tsx:
```diff
@@ -0,0 +1,42 @@
+
+			<Card>
+				<div className="mailpoet-upsell__icon">
+					<img src={ mailpoetSymbol } alt="MailPoet" />
```
Maybe the alternative text could be more specific?

```suggestion
					<img src={ mailpoetSymbol } alt="MailPoet logo" />
```

fredrikekelund replied:
Sure! I suppose that's more in line with the symbols we show for Google/WordPress.

fredrikekelund committed:
Update alt text for MailPoet logo

Co-authored-by: Stéphane Thomas <email@stephanethomas.com>

stephanethomas commented:
What if the user already has enabled this plugin? Shouldn't we hide this upsell if that's the case? Regarding the look-and-feel of this upsell, shouldn't we follow design patterns we already use in other sections? Here is an example of what I'm referring to:

![image](https://user-images.githubusercontent.com/594356/183610258-a4b50dc9-2a4b-4d13-86ed-465247d8b081.png)

fredrikekelund commented:
> What if the user already has enabled this plugin? Shouldn't we hide this upsell if that's the case?

Excellent point. I'll look into that.

> Regarding the look-and-feel of this upsell, shouldn't we follow design patterns we already use in other sections?

We haven't gotten any specific design feedback on the upsell. I would be cautious about adding something large (like the topmost box in your screenshot), especially if people can't close it.

I suppose we could also pursue a larger, more banner-like appearance, but I would prefer to get a mock-up from design to work off of in that case.

stephanethomas commented:
> I suppose we could also pursue a larger, more banner-like appearance, but I would prefer to get a mock-up from design to work off of in that case.

Maybe @semihakocer or @wensco could chime in, and share how far we should go in term of design for that upsell? I agree that the upsell for Yoast is too big 😀. We could also keep it like you did but remove the `Newsletters` header - so it's a bit different from the other sections:

![image](https://user-images.githubusercontent.com/594356/183615009-2692036b-a415-4f03-b829-f4bfc010c415.png)

fredrikekelund committed:
Hide MailPoet upsell if plugin is already installed

fredrikekelund committed:
Adjusted styling for MailPoet upsell, EmailListInactive

fredrikekelund commented:
> We could also keep it like you did but remove the Newsletters header - so it's a bit different from the other sections

I like that! I made that change and increased the top margin.

I also took the opportunity to address some style inconsistencies between `EmailListActive` and `EmailListInactive` (see the difference in height for the `SectionHeader` for the two lists in Stéphane's previous screenshot).

fredrikekelund commented on client/my-sites/email/email-management/home/mailpoet-upsell.tsx:
```diff
@@ -0,0 +1,51 @@
+							<h4>MailPoet</h4>
+							<span>
+								{ translate(
+									'Create and send beautiful newsletters, post notifications, welcome emails and more from WordPress.com.'
```
We discussed some different alternatives for this text blurb previously. I'm open to changing it if @semihakocer or @wensco have suggestions, but I think we should aim for something that fits on a single row on a 13" laptop monitor.

semihakocer replied:
@fredrikekelund what's the character limit to fit in a single line? We can modify the text accordingly.

fredrikekelund replied:
It's not an exact character limit since different characters can render at different widths, but it's somewhere around **120 characters**.

If you use the [Calypso live link](https://calypso.live/?image=registry.a8c.com/calypso/app:build-39420) and then follow the Testing instructions in the PR description, you can verify that a particular string looks good on laptop, if you like (by changing the text through the browser dev tools).

semihakocer replied:
Some alternatives for you to choose from:

Create and send beautiful newsletters, post notifications, and more with MailPoet.

Send beautiful newsletters, post notifications, emails, and more with MailPoet.

Design and send newsletters to your followers with MailPoet.

fredrikekelund replied:
Sorry for the back and forth, but since we were able to fit more copy with the `Banner` format, I riffed off your original suggestion (pbLl1t-1cg-p2#comment-1015). Here's the current heading and subtext:

> MailPoet – Send beautiful newsletters from WordPress

> Collect subscribers, manage your lists and create automatic emails to grow your email marketing reach.

Feedback welcome!

semihakocer replied:
Looks good

fredrikekelund commented on client/my-sites/email/email-management/style.scss:
```diff
@@ -120,18 +122,34 @@
 	margin-top: 15px;
+}
 
-	> .card {
```
This rule would also target the `SectionHeader` in `EmailListInactive`, which it shouldn't. That's why I removed it and changed around the styling a little bit.

stephanethomas commented:
> I also took the opportunity to address some style inconsistencies between EmailListActive and EmailListInactive (see the difference in height for the SectionHeader for the two lists in Stéphane's previous screenshot).

Wow that's cool! This difference has been bothering me for some time but it seemed it would only be visible when stylesheets are loaded in a specific order. I'm glad you've fixed that!

fredrikekelund commented:
For reference, here's what the upsell looks like now:

![mailpoet-upsell](https://user-images.githubusercontent.com/1101677/183625659-0bb913bd-b922-466e-bbb8-c1d89ca90a3b.png)

semihakocer commented:
> Maybe @semihakocer or @wensco could chime in, and share how far we should go in term of design for that upsell? I agree that the upsell for Yoast is too big 😀. We could also keep it like you did but remove the Newsletters header - so it's a bit different from the other sections:

The new design looks better. We can use the JITMs but I think we might want to reserve them for our promotions instead of an evergreen Newsletter highlight. It might make sense to ask design to see if we can highlight this part in a different way.  (More JITM design could be found here: https://mc.a8c.com/jitms/) Also check this component please: https://wpcalypso.wordpress.com/devdocs/blocks/upsell-nudge

fredrikekelund commented:
I'll throw something together with an `UpsellNudge` or `Banner` component so that we have something to compare with!

I think it might be sound to visually distinguish between the upsell and the domains.

wensco commented:
How will we handle displaying this upsell for users who do not have a plan that supports installing plugins? Currently users need to have Business, eCommerce, or the now-legacy Pro plan to install their own plugins.

fredrikekelund commented:
With a `Banner` instead of the current implementation, we would get something like this. I opted to move the upsell nudge to the top, since that felt more natural with this format.

Thoughts or feedback?

![mailpoet-upsell-nudge](https://user-images.githubusercontent.com/1101677/183677821-7ac7c0c2-8075-4ddb-ae59-a97db6a58d04.png)

fredrikekelund commented:
> How will we handle displaying this upsell for users who do not have a plan that supports installing plugins?

It might be smart to only show the upsell to users on plans that support plugins...

The CTA URL will technically work for all users, regardless of what plan they are on and I guess you could argue that this might be an argument for users to upgrade their plan. But my suspicion is that the average user might be somewhat confused about the fact that they have to purchase a new WordPress.com plan when they only meant to install the newsletter solution that we showed them an ad for.

WDYT @wensco?

fredrikekelund committed:
Convert MailPoet upsell to use Banner component

fredrikekelund committed:
Only display MailPoet upsell if install-plugins feature is available

fredrikekelund committed:
Copy adjustments

fredrikekelund commented:
I went ahead and committed the `Banner` approach. It isn't necessary more visually appealing, but I think there are a few clear pros:

- it's a recognizable style across Calypso
- users can close it
- we can fit more copy in there

Here's a screenshot of the latest iteration:

![mailpoet-upsell](https://user-images.githubusercontent.com/1101677/183841290-256ca15b-5c60-43da-87b1-6e2b00d558c9.png)

fredrikekelund commented on client/my-sites/email/email-management/email-home.tsx:
```diff
@@ -121,7 +112,14 @@ const EmailHome = ( props: EmailManagementHomeProps ) => {
-		return <LoadingPlaceholder />;
+		return (
+			<ContentWithHeader>
+				<QueryJetpackPlugins siteIds={ [ selectedSite?.ID ?? 0 ] } />
```
I inlined `LoadingPlaceholder` to make it easier to add the `QueryJetpackPlugins` component here. The reason I wanted to do that was to preload the plugins list before the `MailPoetUpsell` component renders.

fredrikekelund commented on client/my-sites/email/email-management/email-home.tsx:
```diff
@@ -197,26 +195,26 @@ const EmailHome = ( props: EmailManagementHomeProps ) => {
 
 	return (
 		<ContentWithHeader>
-			<>
```
This is just a bit of housekeeping. I changed the type for the `children` prop in the `ContentWithHeader` to `ReactNode` (which is what React uses for its default `children` type). With that change, we no longer need a single child in `ContentWithHeader`, which is why I made this change.

stephanethomas commented:
> With a Banner instead of the current implementation, we would get something like this. I opted to move the upsell nudge to the top, since that felt more natural with this format. Thoughts or feedback?

That looks good to me 👍.

semihakocer commented:
> I went ahead and committed the `Banner` approach. It isn't necessary more visually appealing, but I think there are a few clear pros:

Looks great to me!

AllTerrainDeveloper reviewed (commented):
The code looks good to me, but the behavior would be improved in the sense of adding more stats in Tracks. Like how many times the banner was displayed. Don't you think so?

AllTerrainDeveloper reviewed (changes requested):
As discussed would be nice to at least track the visibility of the banner, and as an extra point show it somehow to users who ONLY have one domain, as for those users the banner is not being presented as they are redirected to the email comparison page

fredrikekelund committed:
Added Tracks impression event for MailPoet upsell

fredrikekelund committed:
Move MailPoetUpsell from EmailHome to EmailPlan

fredrikekelund commented:
I think it's a fair assumption that most users will not have more than one custom domain for their site. As @AllTerrainDeveloper pointed out, these users never see the `EmailHome` component, because they are automatically redirected to either the comparison page or the management page.

That's why I went ahead and moved the upsell to the `EmailPlan` component instead (visible at `/email/:domain/manage/:site_slug`). This should make it visible to more users.

Here's a screenshot for reference:

![mailpoet-upsell](https://user-images.githubusercontent.com/1101677/184119248-48e403a7-96b7-407e-b2f2-3805f8e1652f.png)

I also added a Tracks impression event.

fredrikekelund committed:
Simple responsive adjustments

AllTerrainDeveloper commented:
LGTM!

fredrikekelund committed:
Fix Tracks events in MailPoet upsell

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7471770

Thank you @fredrikekelund for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7471770).
